lib/puppet/util/windows/process.rb

```diff
@@ -204,6 +204,37 @@ def elevated_security?
   end
   module_function :elevated_security?
 
+  def windows_major_version
+    ver = nil
+
+    FFI::MemoryPointer.new(OSVERSIONINFO.size) do |os_version_ptr|
+      os_version = OSVERSIONINFO.new(os_version_ptr)
+      os_version[:dwOSVersionInfoSize] = OSVERSIONINFO.size
+
+      result = GetVersionExW(os_version_ptr)
+
+      if result == FFI::WIN32_FALSE
+        raise Puppet::Util::Windows::Error.new(
+          "GetVersionEx(#{os_version_ptr})")
```

Comment:
Just wondering at its utility in the error message, then.  Does `Windows::Error` have a way at getting at the last error code?  I figure that'd be more use than `os_version_ptr`, no?